From 4fd83a6ed5a86e99ccc01909199143884e9aca90 Mon Sep 17 00:00:00 2001
From: Angela Czubak <aczubak@caviumnetworks.com>
Date: Fri, 19 Oct 2018 10:07:00 +0300
Subject: [PATCH 0336/1051] octeontx-pki: add MBOX_PKI_SET_PORT_CONFIG

MBOX_PKI_SET_PORT_CONFIG is intended to be used for setting additional
port config, such as parsing mode. To make it more useful in packet
instruction header case, mbox_pki_port_cfg was extended with inst_skip
field so that PKI_CL*_PKIND()_SKIP[INST_SKIP] can be modified.

Signed-off-by: Angela Czubak <aczubak@caviumnetworks.com>
Signed-off-by: Yury Norov <ynorov@marvell.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h
index e09f5af7247e..36248244d99c 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_mbox.h
@@ -593,6 +593,7 @@ struct __attribute__((__packed__)) mbox_tim_ring_conf {
 #define MBOX_PKI_GET_PORT_QOS_CONFIG		20
 #define MBOX_PKI_PORT_ALLOC_QPG			21
 #define MBOX_PKI_PORT_FREE_QPG			22
+#define MBOX_PKI_SET_PORT_CONFIG		23
 
 /* pki pkind parse mode */
 enum  {
@@ -622,6 +623,7 @@ typedef struct mbox_pki_port_cfg {
 		/* modify mask 1=modify 0=dont modify*/
 		u8 fcs_pres:1;
 		u8 fcs_skip:1;
+		u8 inst_skip:1;
 		u8 parse_mode:1;
 		u8 mpls_parse:1;
 		u8 inst_hdr_parse:1;
@@ -632,6 +634,7 @@ typedef struct mbox_pki_port_cfg {
 	} mmask;
 	u8 fcs_pres;
 	u8 fcs_skip;
+	u8 inst_skip;
 	u8 parse_mode;
 	u8 mpls_parse;
 	u8 inst_hdr_parse;
-- 
2.17.1

